NEWS.md

```diff
@@ -434,6 +434,11 @@ Library improvements
   * The function `thisind(s::AbstractString, i::Integer)` returns the largest valid index
     less or equal than `i` in the string `s` or `0` if no such index exists ([#24414]).
 
+  * `String(array)` now accepts an arbitrary `AbstractVector{UInt8}` and "steals" the
+    memory buffer of mutable arrays, leaving the byte vector with an empty buffer which
+    is guaranteed not to be shared with the `String` object; if the byte vector is
+    immutable, it simply shares memory with the string and is not truncated ([#26093]).
```

Comment:
This promises too much --- we don't have the ability to share arbitrary immutable vector memory with strings. It also doesn't steal the memory of arbitrary mutable arrays, only `Vector{UInt8}`.

Reply:
Can you suggest a wording?

Reply:
I've proposed something, let me know what you think.